components/script/dom/htmlelement.rs

```diff
@@ -783,6 +786,49 @@ impl HTMLElement {
 
         None
     }
+
+    // https://html.spec.whatwg.org/multipage/#the-summary-element:activation-behaviour
+    pub fn summary_activation_behavior(&self) {
+        // Step 1
+        if !self.is_summary_for_its_parent_details() {
+            return;
+        }
+
+        // Step 2
+        let parent_details = self.upcast::<Node>().GetParentNode().unwrap();
+
+        // Step 3
+        parent_details
+            .downcast::<HTMLDetailsElement>()
+            .unwrap()
+            .toggle();
+    }
+
+    // https://html.spec.whatwg.org/multipage/#summary-for-its-parent-details
+    fn is_summary_for_its_parent_details(&self) -> bool {
+        // Step 1
+        let summary_node = self.upcast::<Node>();
+        if !summary_node.has_parent() {
+            return false;
+        }
+
+        // Step 2
+        let parent = &summary_node.GetParentNode().unwrap();
+
+        // Step 3
+        if parent.downcast::<Element>().unwrap().local_name() != &local_name!("details") {
```

Comment:
This could be replaced by `parent.is::<HTMLDetailsElement>()` instead.

Reply:
Yep, much simpler to read, thanks :)